pytorch_lightning/plugins/precision/apex_amp.py

```diff
@@ -86,6 +92,15 @@ def backward(
         closure_loss = closure_loss.detach()
         return closure_loss
 
+    def pre_optimizer_step(
+        self, pl_module: LightningModule, optimizer: Optimizer, optimizer_idx: int, closure: Callable, **kwargs
+    ) -> bool:
+        """Hook to do something before each optimizer step."""
+        # Apex: Amp does not support closure use with optimizers
+        closure()
+        optimizer.step()
+        return False
```

Comment:
This doesn't need to return false, but stays consistent with the native_amp plugin.

The bool represents whether to run to run the optimizer step or not. in the case of native_amp, we do `scaler.step(optimizer)`.

I don't like this at all, however because of the disjoint of responsibility, this has to be remain as is. I plan to address the responsibility of precision in a separate RFC after 1.2